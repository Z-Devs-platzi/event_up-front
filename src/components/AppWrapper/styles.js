import styled from 'styled-components';

export const Wrapper = styled.div`
  display: grid;
  width: 100vw;
  height: 100vh;
  grid-template-rows: 90px auto;
  grid-template-columns: 248px auto;
  grid-template-areas:
    'nav header'
    'nav content';
  @media screen and (max-width: 900px) {
    grid-template-rows: 56px 90px auto 60px;
    grid-template-columns: auto;
    grid-template-areas:
      'nav'
      'header'
      'content'
      'foot';
  }
`;
export const WrapperContent = styled.div`
  grid-area: content;
  padding: 5px;
`;
export const WrapperHeader = styled.div`
  grid-area: header;
`;
export const NavContainer = styled.div`
  grid-area: nav;
`;
