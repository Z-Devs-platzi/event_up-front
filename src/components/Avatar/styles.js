import styled from 'styled-components';

export const AvatarImage = styled.img`
  --size: ${(props) => props.size};
  display: block;
  width: var(--size);
  height: var(--size);
  background-color: var(--grey);
  border-radius: 50%;
  object-fit: cover;
  border: none;
`;
