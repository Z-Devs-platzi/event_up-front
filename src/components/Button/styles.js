import styled from 'styled-components';

export const Buttons = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 1em;
  padding: 0.5rem 2rem;
  background: ${(props) =>
    props.buttonType === 'primary' ? 'var(--secondary-color)' : 'transparent'};
  border: ${(props) =>
    props.buttonType === 'primary'
      ? '1px solid var(--secondary-color)'
      : '1px solid var(--black)'};
  color: var(--black);
  cursor: pointer;
  font-size: 1rem;
  font-weight: 700;
  border-radius: 0.5rem;
  transition: background 0.3s ease-in-out;
  &:hover {
    background: ${(props) =>
      props.buttonType === 'primary'
        ? 'var(--white)'
        : 'var(--secondary-color)'};
  }
`;
