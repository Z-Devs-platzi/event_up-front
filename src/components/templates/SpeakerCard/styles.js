import styled from 'styled-components';

export const SpeakerWrapper = styled.div`
  display: flex;
  align-items: center;
  margin: 1.25rem 0;

  @media (max-width: 480px) {
    flex-direction: column;
    justify-content: center;
  }
`;

export const SpeakerInfoWrapper = styled.div`
  margin-left: 1.5rem;
  @media (max-width: 480px) {
    text-align: center;
    margin-left: 0;
    margin-top: 0.5rem;
  }
`;

export const SpeakerName = styled.h4`
  position: relative;
  font-size: 1.5rem;
  font-family: var(--template-font), sans-serif;
  color: var(--template-color-secondary);
  &::after {
    content: '';
    position: absolute;
    bottom: -2px;
    left: 0;
    display: block;
    width: 3rem;
    background-color: var(--template-color-secondary);
    height: 0.06rem;
  }
  @media (max-width: 480px) {
    &::after {
      right: 0;
      margin: 0 auto;
    }
  }
`;

export const SpeakerBio = styled.p`
  margin: 1.25rem 0 0.625rem;
  font-family: var(--template-font), sans-serif;
  color: var(--black);
  font-size: 1rem;
`;

export const SpeakerTwitter = styled.a`
  display: flex;
  align-items: center;
  font-family: var(--template-font), sans-serif;
  color: var(--template-color-secondary);
  font-size: 0.75rem;
  text-decoration: underline;
  column-gap: 0.25rem;
  @media (max-width: 480px) {
    justify-content: center;
  }
`;
