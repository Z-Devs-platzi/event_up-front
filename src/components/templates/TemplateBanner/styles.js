import styled from 'styled-components';

export const BannerWrapper = styled.section`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 60vh;
  background-color: var(--grey);
  overflow: hidden;
`;

export const ImageContainer = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  width: 100%;
  height: 100%;
  background-image: url(${(props) => props.bgImage});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  filter: grayscale(100%);
`;

export const OverlayLayer = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0.5;
  z-index: 2;
  background: linear-gradient(
    180deg,
    rgba(18, 94, 104, 0.85) 0%,
    rgba(255, 212, 59, 0.13) 152%
  );
  background-blend-mode: overlay;
  width: 100vw;
  height: 100%;
`;

export const InfoContainer = styled.div`
  position: relative;
  z-index: 3;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
`;

export const EditionTools = styled.div`
  position: absolute;
  top: 0;
  right: 0;
`;
