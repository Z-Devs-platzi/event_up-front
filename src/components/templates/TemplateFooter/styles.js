import styled from 'styled-components';

export const FooterWrapper = styled.footer`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding: 2.5rem;
  background-color: var(--template-color-secondary);
  box-sizing: border-box;
`;

export const LogoContainer = styled.div`
  color: var(--white);
  font-size: 0.75rem;
  text-align: center;
`;

export const Logo = styled.img`
  display: block;
  width: 5.5rem;
`;
