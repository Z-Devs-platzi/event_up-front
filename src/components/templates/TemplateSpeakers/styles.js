import styled from 'styled-components';

export const TemplateSpeakersWrapper = styled.section`
  display: flex;
  flex-direction: column;
  padding: 1.25rem 3.5rem;
  background-color: var(--template-color-compA);
  @media (max-width: 480px) {
    padding: 1.25rem 1rem;
  }
`;

export const TemplateSpeakersTitle = styled.h2`
  font-family: var(--template-font), sans-serif;
  font-size: 6rem;
  color: var(--template-color-secondary);
  opacity: 0.2;
  @media (max-width: 480px) {
    font-size: 3.5rem;
    text-align: center;
  }
`;
