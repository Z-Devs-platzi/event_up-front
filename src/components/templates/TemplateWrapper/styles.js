import styled from 'styled-components';

export const Wrapper = styled.main`
  --template-color-primary: ${(props) => props.primary};
  --template-secondary: ${(props) => props.secondary};
  --template-compA: ${(props) => props.compA};
  --template-compB: ${(props) => props.compB};
  --template-font: ${(props) => props.font};
`;
