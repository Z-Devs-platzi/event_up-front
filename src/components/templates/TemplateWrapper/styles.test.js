import React from 'react';
import { Wrapper } from './styles';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('templates / templateWraper', () => {
  it('templateWraper', () => {
    const timerStyles = renderer.create(<Wrapper />).toJSON();
    expect(timerStyles).toMatchSnapshot();
  });
});
