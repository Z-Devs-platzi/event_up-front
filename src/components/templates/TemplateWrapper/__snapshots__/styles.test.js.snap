// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates / templateWraper templateWraper 1`] = `
<main
  className=""
/>
`;
